psqlgraph/node.py

```diff
@@ -240,13 +250,12 @@ def to_json(self):
 
     @classmethod
     def from_json(cls, node_json):
-        if cls is Node:
-            Type = Node.get_subclass(node_json['label'])
-            if not Type:
-                raise KeyError('Node has no subclass named {}'
-                                  .format(node_json['label']))
-        else:
-            Type = cls
+        abstract_node_cls = cls.get_edge_class().get_node_class()
+        Type = abstract_node_cls.get_subclass(node_json["label"])
+
+        if not Type:
```

Comment:
I feel like the new implementation will hide a potentiall issue with calls like:

```
from gdcdatamodels.models import TypeA

TypeA.from_json({'label': 'type_b', ...})
```
where we'd automatically resolve to a `TypeB` node and hide the fact that client didn't provide correct class

Reply:
L253 here is to ensure the load is coming from the proper base class, if `type_b` does not exists in the base class of `TypeA` the exception will be raised. Also, only the default base classes will be loaded into `gdcdatamodels.models`, the namespaced ones will be loaded into `gdcdatamodels.models.namespace`

Reply:
ok, in my snippet above I meant:
```
from namespace import TypeA, TypeB

TypeA.from_json({'label': 'type_b', ... })
```
so, both `TypeA` and `TypeB` exist in the same namespace. old behavior wouldn't allow the instantiation to happen, but the new behavior will get base class and then get `TypeB` subclass and create instance, which I think is not what we want to happen. we want it to fail

Reply:
I see your point, I missed out the `if cls != Node` logic. Good eyes